Ramanujan_joel.java

```diff
@@ -0,0 +1,52 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by Fernflower decompiler)
+//
+
+public class Ramanujan {
+  public Ramanujan() {
+  }
+
+  private static boolean isWholeNumber(double error, double n) {
+    return error > Math.abs(n - Math.ceil(n)) || error > Math.abs(n - Math.floor(n));
+  }
+
+  public static boolean isRamanujan(long n) {
+    double error = 1.0E-11D;
```

Comment:
doubles have finite precision.  depending on your application this like dealing with money, this will not be good.

Reply:
Yeah, I put a note on this problem that I need to re-think how I get my whole numbers and determine if a value is a whole number.

Reply:
also, I failed the test cases for an extremely large number, so they got me there too.